### DISCUSSION TITLE
Current method makes toggling Copilot on/off cumbersome

### DISCUSSION THREAD

**Discussion by alirezaghey** [community/community#8433] [Copilot Conversations]
At times I need to toggle Copilot on/off quickly. Currently, even if you set a handy keyboard shortcut for it, Copilot opens a dialog asking if you want to disable Copilot globally or for a specific language. This forces you to leave the keyboard and manually choose an options.

It would be great if we just could set a default for Copilot's behavior on enabling/disabling and then toggle it on and off swiftly with a shortcut. Maybe it would be sensible to have 2 different commands: One for toggling Copilot on/off globally and the other for the specific language you are working on.

Anyway, thank you for a great product.

**Comment by eli-crow** [comment]
I work in an environment where clients may be very sensitive about any sharing of code, however securely. Currently, the enable/disable setting modifies my settings file, meaning every time I re-enable it, "enabled" becomes the default again. But I'd like a more temporary toggle, so I can opt in when working on personal projects.

1. Disable globally by default
2. Enable for the current session
3. Reverts to disabled again next time I open Code.
